### PR TITLE
Add forced notebook lock release

### PR DESCRIPTION
**Pull request by jlewi**
## Summary

- add a notebook-targeted BroadcastChannel protocol that lets the live Web Lock owner self-identify and cooperatively release its lease
- cancel active runner, AppKernel, and Jupyter executions; flush pending changes before release; and keep the former owner open in locked/read-only mode
- let the requester retry normal acquisition, surface timeout/failure states without automatic retries, and reload the former owner after the new writer acquires
- document the mechanism and resolved design decisions in `docs-dev/design/20260702_forced_notebook_lock_release.md`
- add a shared Excalidraw test boundary mock so the complete Vitest suite can run in Node without loading its canvas/browser bundle

## Why

A notebook can currently be edited in only one browser session, but a lost or inaccessible session can retain the write lock. Stale IndexedDB owner metadata cannot safely identify that owner, so force-release requests are broadcast by notebook URI and only the session holding the authoritative Web Lock responds.

## Validation

- `corepack pnpm -C app exec vitest run` (72 files, 639 tests)
- `runme run build test`
- ESLint on all changed source files (0 errors)
- two-session browser verification:
  - requester became editable and former owner became read-only
  - debounced edits were saved during takeover
  - the former owner reloaded persisted content with editing controls disabled

### CODE REVIEW

File: app/src/lib/notebookDataController.ts [hunk 1]
```diff
@@ -290,6 +328,199 @@ export class NotebookDataController {
...
+      } catch (error) {
+        handle.data.setReleasePending(false)
+        handle.data.setReadOnly(false)
+        handle.data.setNotebookStore(this.createOwnedNotebookStore(localUri))
+        const currentEntry =
+          this.openNotebooks.find((item) => item.uri === localUri) ?? entry
+        this.upsertOpenEntry({
+          ...currentEntry,
+          state: 'loaded',
+          readOnly: false,
```
[the remark this answers is not in the document]
Comment (jlewi-openai) [reply app/src/lib/notebookDataController.ts]: Addressed in d039213. The forced-release path now restores writable state only when the same open entry, handle, and lease still exist, and it exits without recreating state when the notebook was closed or reopened. Added regression coverage for both successful and failed pending saves.
